autogpt/main.py

```diff
@@ -186,19 +188,212 @@ def run_auto_gpt(
         "Using memory of type:", Fore.GREEN, f"{memory.__class__.__name__}"
     )
     logger.typewriter_log("Using Browser:", Fore.GREEN, config.selenium_web_browser)
-    system_prompt = ai_config.construct_full_prompt(config)
-    if config.debug_mode:
-        logger.typewriter_log("Prompt:", Fore.GREEN, system_prompt)
 
     agent = Agent(
-        ai_name=ai_name,
         memory=memory,
-        next_action_count=next_action_count,
         command_registry=command_registry,
-        system_prompt=system_prompt,
         triggering_prompt=DEFAULT_TRIGGERING_PROMPT,
         workspace_directory=workspace_directory,
         ai_config=ai_config,
         config=config,
     )
-    agent.start_interaction_loop()
+
+    run_interaction_loop(agent)
+
+
+SignalHandler = Callable[[int, Optional[FrameType]], None]
+
+
+def graceful_agent_interrupt(
+    agent: Agent,
+    logger_: Logger,
+) -> SignalHandler:
+    """Create a signal handler to interrupt an agent executing multiple steps.
+
+    This is used to allow the agent to finish its current step before exiting.
+
+    Args:
+        agent: The agent to interrupt.
+        logger_: The logger to use to print a message.
+
+    Returns:
+        A signal handler that can be used to interrupt the agent.
+    """
+
+    def signal_handler(signum: int, frame: Optional[FrameType]) -> None:
+        if agent.cycle_budget == 0 or agent.cycles_remaining == 0:
+            sys.exit()
+        else:
+            logger_.typewriter_log(
+                "Interrupt signal received. Stopping continuous command execution.",
+                Fore.RED,
+            )
+            agent.cycles_remaining = 0
```

Comment:
Doesn't seem to work properly:
>   Cycle budget: 5; remaining: 5
>   Creating chat completion with model gpt-3.5-turbo, temperature 0.0, max_tokens 1023

`Ctrl`+`C`

> Aborted!
> Press any key to continue...

Reply:
fixed the issue on https://github.com/Significant-Gravitas/Auto-GPT/pull/4997

Reply:
Thank you! :)